tests/test_database.py

```
@@ -93,6 +104,34 @@ def test_run_startup_migrations_with_existing_tables_no_alembic_version(
+@patch("alembic.command.stamp")
+@patch("app.database.engine")
+def test_run_startup_migrations_stamps_max_concurrent_schema_to_matching_revision(
+    mock_engine, mock_stamp, mock_upgrade, db_session, monkeypatch
+):
+    """Databases with the column already present should not be re-migrated from the base revision."""
+    monkeypatch.setattr("app.config.settings.database_path", ":memory:")
+
+    mock_inspector = MagicMock()
+    mock_inspector.get_table_names.return_value = ["monitored_paths", "users"]
+
+    def get_columns(table_name):
```

Ruff (0.15.9)

[warning] Missing return type annotation for private function `get_columns`

(ANN202)